core-common/src/main/java/org/mapstruct/MapMapping.java

```diff
@@ -55,6 +55,19 @@
     String valueDateFormat() default "";
 
     /**
+     * A qualifier can be specified to aid the selection process of a suitable mapper. This is useful in case multiple
+     * mappers (hand written of internal) qualify and result in an 'Ambiguous mapping methods found' error. A qualifier
+     * is a custom annotation and can be placed on either a hand written mapper class or a method.
+     * <p>
+     * Qualifiers listed here are used for the selection of method-level aspects, and in case nothing is specified for
```

Comment:
I forgot to change some of that back again. Originally, I had implemented that in case no key/value qualifiers were given, then qualifiers from this property are taken for them as well. I removed that part again and forgot to adjust the Javadoc.
The qualifiedBy method itself was added to have a place for qualifiers that are taken into account for selecting before/after methods.